optuna/study.py

```diff
@@ -538,6 +554,21 @@ def _optimize_sequential(
             self._run_trial_and_callbacks(func, catch, callbacks, gc_after_trial)
         self._storage.remove_session()
 
+    def _pop_waiting_trial_id(self):
+        # type: () -> Optional[int]
+
+        for trial in self.trials:
+            if trial.state != structs.TrialState.WAITING:
+                continue
+
+            if not self.storage.set_trial_state(trial.trial_id, structs.TrialState.RUNNING):
+                continue
+
+            self.logger.debug("Trial#{} is popped from the trial queue.".format(trial.number))
+            return trial.trial_id
```

Comment:
I think this change introduces some room for performance improvements, at least for the RDB storage, namely to introduce a DB query for extracting waiting trials instead of looping over all of them. Could you leave a TODO comment?